llvm/include/llvm-c/CAS/PluginAPI_functions.h

```diff
@@ -0,0 +1,244 @@
+/*===-- llvm-c/CAS/PluginAPI_functions.h - CAS Plugin Functions Interface -===*\
+|*                                                                            *|
+|* Part of the LLVM Project, under the Apache License v2.0 with LLVM          *|
+|* Exceptions.                                                                *|
+|* See https://llvm.org/LICENSE.txt for license information.                  *|
+|* SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception                    *|
+|*                                                                            *|
+|*===----------------------------------------------------------------------===*|
+|*                                                                            *|
+|* The functions for the LLVM CAS plugin API. Intended for assisting          *|
+|* implementations of the API.                                                *|
+|*                                                                            *|
+\*===----------------------------------------------------------------------===*/
+
+#ifndef LLVM_C_CAS_PLUGINAPI_FUNCTIONS_H
+#define LLVM_C_CAS_PLUGINAPI_FUNCTIONS_H
+
+#include "llvm-c/CAS/PluginAPI_types.h"
+#include "llvm-c/ExternC.h"
+
+#ifndef LLCASAPI_PUBLIC
+#ifdef _WIN32
+#define LLCASAPI_PUBLIC __declspec(dllexport)
+#else
+#define LLCASAPI_PUBLIC
+#endif
+#endif
+
+LLVM_C_EXTERN_C_BEGIN
+
+/**
+ * Returns the \c LLCASAPI_VERSION_MAJOR and \c LLCASAPI_VERSION_MINOR values
+ * that the plugin was compiled with.
+ * Intended for assisting compatibility with different versions.
+ */
+LLCASAPI_PUBLIC void llcasapi_get_plugin_version(unsigned *major,
```

Comment:
My rationale for picking `llcasapi` is because this is intended as the "API contract" for a plugin to follow, and I wanted to avoid potential confusion if in the future `llvm-c` starts providing a C API for the `LLVMCAS` library itself.

But I don't have strong opinion, if both you and @benlangmuir prefer `llcas` I'll change it.

Reply:
To make sure I understand, the idea is that if you are working with a CAS in the context of LLVM you would use the (hypothetical) API from llvm-c `LLVMCAS...` and this `llcasapi_` is only for the plugins?

Personally, I don't find `llcas_` confusing, since it's a completely different naming scheme from LLVM C API which is upper-camel case, and will be "LLVM" not just "LL".

But I also don't feel that strongly about it.

Reply:
I changed it to `llcas` prefix.